zabbix/files/default/etc/zabbix/zabbix_agentd.conf.jinja

```diff
@@ -189,8 +189,9 @@ ServerActive={{ settings.get('serveractive', defaults.serveractive) }}
 # Mandatory: no
 # Default:
 # Hostname=
-
+{% if not settings.get('hostnameitem', defaults.get('hostnameitem', False)) -%}
 Hostname={{ settings.get('hostname', defaults.get('hostname', salt['grains.get']('id'))) }}
+{% endif %}
 
 ### Option: HostnameItem
```

Comment:
Sorry, but that is not exactly what I suggested, with this change users still will get unnecessary changes in their config files after formula update.

My suggestion was

```jinja
# Hostname=
{% if not settings.get('hostnameitem', defaults.get('hostnameitem', False)) %}
Hostname={{ settings.get('hostname', defaults.get('hostname', salt['grains.get']('id'))) }}
{% endif %}
### Option: HostnameItem
```

Note, there is `%}` not `-%}` in `if` opening block and also no extra empty line before `### Option: HostnameItem`. We want, when possible, that users won't get changes after simple formula update, that's why I'm so picky about every line and even symbol.

Commits message is good now.

Reply:
Sorry for that. 
I try different things because of the failed tests. But they are not related to my changes.